airbyte-integrations/connectors/source-recharge/integration_tests/abnormal_state.json

```diff
@@ -2,49 +2,49 @@
   {
     "type": "STREAM",
     "stream": {
-      "stream_state": { "updated_at": "2050-05-18T00:00:00" },
+      "stream_state": { "updated_at": "2050-05-18T00:00:00Z" },
       "stream_descriptor": { "name": "addresses" }
     }
   },
   {
     "type": "STREAM",
     "stream": {
-      "stream_state": { "updated_at": "2050-05-18T00:00:00" },
+      "stream_state": { "updated_at": "2050-05-18T00:00:00Z" },
       "stream_descriptor": { "name": "charges" }
     }
   },
   {
     "type": "STREAM",
     "stream": {
-      "stream_state": { "updated_at": "2050-05-18T00:00:00" },
+      "stream_state": { "updated_at": "2050-05-18T00:00:00Z" },
       "stream_descriptor": { "name": "customers" }
     }
   },
   {
     "type": "STREAM",
     "stream": {
-      "stream_state": { "updated_at": "2050-05-18T00:00:00" },
+      "stream_state": { "updated_at": "2050-05-18T00:00:00Z" },
       "stream_descriptor": { "name": "discounts" }
     }
   },
   {
     "type": "STREAM",
     "stream": {
-      "stream_state": { "updated_at": "2050-05-18T00:00:00" },
+      "stream_state": { "updated_at": "2050-05-18T00:00:00Z" },
       "stream_descriptor": { "name": "onetimes" }
     }
   },
   {
     "type": "STREAM",
     "stream": {
-      "stream_state": { "updated_at": "2050-05-18T00:00:00" },
+      "stream_state": { "updated_at": "2050-05-18T00:00:00Z" },
       "stream_descriptor": { "name": "orders" }
     }
   },
   {
     "type": "STREAM",
     "stream": {
-      "stream_state": { "updated_at": "2050-05-18T00:00:00" },
+      "stream_state": { "updated_at": "2050-05-18T00:00:00Z" },
```

Comment:
Not sure why the records for the incremental streams were emitted with the `"updated_at": "2050-05-18T00:00:00"`, since:

Original RECORD example: (partially provided here to save place)
```json
{
"subscriptions": [
        {
            ....
            "id": 153612006,
            "sku_override": false,
            "status": "cancelled",
            **"updated_at": "2022-03-22T21:26:07+00:00",**
            "variant_title": "One Size / Black"
            ....
        }
]
```

The data goes with the `TZ` in this format: `- "%Y-%m-%dT%H:%M:%S%z"` and is being emitted as is, satisfying the record data format.

The tests brought me to the `2021-01` API version, where the was no `TZ` component: `'2022-03-22T17:26:07'`, and because the `future_state` checks are disabled for the `CAT` the `abnormal_state.json` was not even triggered for the tests. I've discovered this manually actually and made changes up to the fact we still want to use the `modern API == 2021-11` for every stream we can.

Conclusion: 
the `abnormal state values` for the existing version were based on the `2021-01` (old) API version, instead of `2021-11` (modern API). Thus the change makes sense to me. WDYT?

Reply:
I've ensured the `old` STATE format of: `{ "updated_at": "2023-02-18T00:00:00" }` is suitable for the migrated version, by adding the additional format to the `cursor_datetime_formats`, it does the thing.

As for the migrated STATE fromat:
I've run the
```
docker run --rm -v $(pwd)/secrets:/secrets -v $(pwd)/integration_tests:/integration_tests airbyte/source-recharge:1.1.6 read --config /secrets/config.json --catalog /integration_tests/configured_catalog.json --state /integration_tests/state.json
```
with the next states:
1) Old format:
```json
{
    "subscriptions":{
        "updated_at": "2023-02-18T00:00:00"
    }
}
```
2) New format:
```json
{
    "subscriptions":{
        "updated_at": "2023-02-18T00:00:00+00:00"
    }
}
```
Both of these are worked out with the migrated and with the old versions.
We should be safe here.

Reply:
Cool! Thanks for validating. Not being a breaking change is a blessing!